docs/api/shallow.md

```diff
@@ -50,6 +50,7 @@ describe('<MyComponent />', () => {
   - `options.disableLifecycleMethods`: (`Boolean` [optional]): If set to true, `componentDidMount`
 is not called on the component, and `componentDidUpdate` is not called after
 [`setProps`](ShallowWrapper/setProps.md) and [`setContext`](ShallowWrapper/setContext.md).
+  - `options.lifecycleExperimental`: (`Boolean` [optional]): If set to true, the entire lifecycle (`componentDidMount` and `componentDidUpdate`) of the React component is called
```

Comment:
Let's also note that the current default is `false`; the next major version will flip the default to `true`; and setting an explicit `true` or `false` here will mean the default change will not effect you.